### ISSUE TITLE
Fresh install of NestJS global CLI and scaffolding a new yarn project fails

### ISSUE THREAD

**Issue by lirantal** [nestjs/nest#8942]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

The global nest CLI is installed without an issue, using:
```
yarn global add @nestjs/cli
```

Then scaffolding a project using the following command:
```
nest new demo123
```

displays the files to be created in that `demo123/` folder, I then continued to choose `yarn` as the package manager, and after some time it failed with the following:

```
? Which package manager would you ❤️  to use? yarn
▹▹▹▹▸ Installation in progress... ☕
Failed to execute command: yarn install --silent
✖ Installation in progress... ☕
🙀  Packages installation failed, see above

                          Thanks for installing Nest 🙏
                 Please consider donating to our open collective
                        to help us maintain this package.


               🍷  Donate: https://opencollective.com/nest
```

### Minimum reproduction code

n/a

### Steps to reproduce

1. `yarn global add @nestjs/cli`
2. `nest new demo123`

### Expected behavior

Project is scaffolded successfully and all dependencies installed properly.

### Package

- [ ] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### NestJS version

8.2.0

### Packages versions

I ran `nest info` and getting the following:

```
 _   _             _      ___  _____  _____  _     _____
| \ | |           | |    |_  |/  ___|/  __ \| |   |_   _|
|  \| |  ___  ___ | |_     | |\ `--. | /  \/| |     | |
| . ` | / _ \/ __|| __|    | | `--. \| |    | |     | |
| |\  ||  __/\__ \| |_ /\__/ //\__/ /| \__/\| |_____| |_
\_| \_/ \___||___/ \__|\____/ \____/  \____/\_____/\___/


[System Information]
OS Version     : macOS Catalina
NodeJS Version : v14.15.1
NPM Version    : 8.1.1

[Nest CLI]
Nest CLI Version : 8.2.0

[Nest Platform Information]
😼  cannot read your project package.json file, are you inside your project directory?
```

However I am running it in the folder which has a `package.json` file in it

### Node.js version

14.15.1

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by lirantal** [comment]
Once the nest cli failed scaffolding, I ran the `yarn install` command manually inside the project and received the following errors, which presumably caused yarn to fail during the `nest new` command:

```
❯ yarn install
yarn install v1.22.10
info No lockfile found.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
error @typescript-eslint/eslint-plugin@5.10.0: The engine "node" is incompatible with this module. Expected version "^12.22.0 || ^14.17.0 || >=16.0.0". Got "14.15.1"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

Looks like the Node.js `v14.15` that I'm running is incompatible with the that specific eslint-plugin requirements, which fails  new scaffolds.

**Comment by micalevisk** [comment]
yeah, the generate project uses the latest version of `@typescript-eslint/eslint-plugin` and they didn't support node v14.15 :disappointed: I guess you'll need to downgrade `@typescript-eslint/*` to v4 https://github.com/typescript-eslint/typescript-eslint/blame/0acfafcc655e28dcfc05a5caa567c0d0217ee7ad/packages/eslint-plugin/package.json#L12

---

btw @lirantal I've read a bunch of thing that you wrote about node :smile: _big fan here_ haha
